test/e2e/app/app.go

```diff
@@ -325,6 +325,9 @@ func (app *Application) ApplySnapshotChunk(_ context.Context, req *abci.RequestA
 // total number of transaction bytes to exceed `req.MaxTxBytes`, we will not
 // append our special vote extension transaction.
 func (app *Application) PrepareProposal(_ context.Context, req *abci.RequestPrepareProposal) (*abci.ResponsePrepareProposal, error) {
+	app.mu.Lock()
```

Comment:
Strictly speaking, this wouldn't be needed, as we're only accessing `app`'s logger, not the state.

But I'm OK with the change, since it makes the method more readable, and there's negligible performance impact (no contention)

Reply:
this mutex is the thing that serializes access to the app, and could contribute to our problems with the abci client. I'm not really worried about potential for data access issues, but more that *not* having a mutex on this means that these calls can run at the same time as other calls.